tracing-appender/README.md

```diff
@@ -0,0 +1,130 @@
+[`tracing`][tracing] is a framework for structured, event-based diagnostic information. `tracing-appender` allows
+events and spans to be recorded in a non-blocking manner through a dedicated logging thread. 
+It also provides a [`RollingFileAppender`][file_appender] that can be used with _or_ without the non-blocking writer.
+
+## Usage
+
+Add the following to your `Cargo.toml`:
+```toml
+tracing-appender = "0.1"
+```
+
+This crate can be used in a few ways to record spans/events:
+ - Using a [`RollingFileAppender`][file_appender] to write to a log file. This is a blocking operation.
+ - Using *any* type implementing [`std::io::Write`][write] in a non-blocking fashion.
+ - Using [`NonBlocking`][non_blocking] and [`RollingFileAppender`][file_appender] together to write to write to log files in a non-blocking fashion.
+
+## Rolling File Appender
+
+```rust
+fn main(){
+    let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
+}
+```
+This creates an hourly rotating file appender that writes to `/some/directory/prefix.log.YYYY-MM-DD-HH`.
+[`Rotation::DAILY`] and [`Rotation::NEVER`] are the other available options.
+
+The file appender implements [`std::io::Write`][write]. To be used with [`tracing_subscriber::FmtSubscriber`][fmt_subscriber], 
+it must be combined with a [`MakeWriter`][make_writer] implementation to be able to record tracing spans/event.
+
+The [rolling] module's documentation provides more detail on how to use this file appender.
+
+## Non-Blocking Writer
+The example below demonstrates the construction of a `non_blocking` writer with `std::io::stdout()`, which implements [`MakeWriter`][[make_writer].
+```rust
+use std::io::Error;
+
+struct TestWriter;
+
+impl std::io::Write for TestWriter {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let buf_len = buf.len();
+    
+        println!("{:?}", buf);
+        Ok(buf_len)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+fn main() {
+    let (non_blocking, _guard) = tracing_appender::non_blocking(TestWriter);
+    let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+    tracing::subscriber::set_global_default(subscriber.finish()).expect("Could not set global default");
+}
+```
+**Note:** `_guard` is a [`WorkerGuard`][guard] which is returned by `tracing_appender::non_blocking`
+to ensure buffered logs are flushed to their output in the case of abrupt terminations of a process.
+See [`WorkerGuard`][guard] module for more details.
+
+The example below demonstrates the construction of a [`tracing_appender::non_blocking`][non_blocking] writer constructed with a [`std::io::Write`][write]:
+```rust
+fn main() {
+    let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::Stdout);
+    let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+    tracing::subscriber::set_global_default(subscriber.finish()).expect("Could not set global default");
+}
```

Comment:
TIOLI: What do you think about using the chaining-based API for setting the subscriber?
```suggestion
```rust
fn main() {
    let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::Stdout);
    tracing_subscriber::fmt()
        .with_writer(non_blocking)
        .init();
}
```